Moving docs from sdcore docs to github repository (#163)
omec-project/gnbsim: docs/deployment.md

<!--
SPDX-FileCopyrightText: 2021 Open Networking Foundation <info@opennetworking.org>
SPDX-License-Identifier: Apache-2.0

-->

# gNBSim deployment options

## Option1: As docker container 

> [Aether onRamp](https://docs.aetherproject.org/master/onramp/gnbsim.html) installs gNBSim as docker container next to SD-Core (5G Core).

Reviewer: ```suggestion
[Aether onRamp](https://docs.aetherproject.org/master/onramp/gnbsim.html) installs gNBSim as docker container 
next to SD-Core (5G Core) or can be deployed in a separate host.
```

If user wish to use just gNBSim and test it with some other 5G Core then Aether onRamp allows just installation of gNBSim. 

> Aether onRamp makefiles are readable and  user can achieve the required goal.

Reviewer: ```suggestion
Aether onRamp makefile files are readable and user can achieve the required goal.
```

## Option2: As standalone executable

> If you want to run gNBSim as a standalone running executable tool then follow below steps.

Reviewer: ```suggestion
If you want to run gNBSim as a standalone executable tool, then follow below steps.
```

>         $ git clone git@github.com:omec-project/gnbsim.git

Reviewer: Any reason you prefer `git`? why not `https`? I think using https is more general/typical because there may be cases where `git` is blocked in corporate networks
```suggestion
        $ git clone https://github.com/omec-project/gnbsim.git
```

        $ cd gnbsim
        $ go build

Run below command to run 5G call flows testing using default configuration file

        $ ./gnbsim

To provide a different configuration file, use the below command

         $ ./gnbsim --cfg config/gnbsim.yaml

> > [!Note] By default, the gNB Sim reads the configuration from ./config/gnbsim.yaml file.

Reviewer: ```suggestion
> [!Note] By default, the gNBSim reads the configuration from ./config/gnbsim.yaml file.
```

> Please refer to the gNBSim configuration [guide](./docs/config.md). Please make sure following
> configuration fields are updated to correct values.

> For each gNB in the config file change below values,

Reviewer: ```suggestion
For each gNB in the config file change below values
```

      n2IpAddr: # gNB N2 interface IP address used to connect to AMF
      n3IpAddr: 192.168.251.5 # gNB N3 interface IP address used to connect to UPF. when singleInterface mode is false

Update amf endpoint to correct FQDN or IP address

      defaultAmf:
        hostName: amf # Host name of AMF
        ipAddr: # AMF IP address
        port: 38412 # AMF port

Update defaultAs value to correct IP address. This address is used to send some test data packets after UE PDU session establishment.

      defaultAs: "192.168.250.1" #default icmp pkt destination

## Option3: As a part of Aether In a Box

> - This is default mode of deployment for gNB Simulator

Reviewer: ```suggestion
- This is default mode of deployment for gNBSim
```

> - Multus cni needs to be enabled on cluster. Required for bess-upf & gNBSim

Reviewer: ```suggestion
- Multus cni needs to be enabled on cluster. Required for bess-upf and gNBSim
```

- `make 5gc` will by default deploy gNB Simulator in this mode
- One interface is used for user plane traffic towards UPF
- Second interface is used to send traffic towards control plane (i.e. AMF).

> - UPF network & default gateway is provided in the override values.

Reviewer: ```suggestion
- UPF network and default gateway is provided in the override values.
```
